### PR TITLE
Update openapi spec

### PR DESCRIPTION
**Pull request by bretterer**
- Updates to latest spec
	(See changelog for details)
- Moved all generated files to root level instead of `Generated` folder and namespace
- Update to allow for use of setAcceptType and setContentType headers
- Updated travis php versions to test
- Changed phpcs (php code style) to ignore line length due to comments

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ - Deleted Object: Okta\Applications/ApplicationTemplateName
+ - Deleted Object: Okta\Applications/JsonWebKeyRSAMediated
+ - Deleted Object: Okta\Applications/PublicSignOnModeMediated
+ - Deleted Object: Okta\Generated/Applications/Collection
+ - Deleted Object: Okta\Generated/UserFactors/Collection
+ - Deleted Object: Okta\Generated/UserFactors/SecurityQuestionFactor
+ - Deleted Object: Okta\Generated/UserFactors/TokenFactor
+ - Deleted Object: Okta\Generated/UserFactors/WebFactor
+ - Deleted Object: Okta\Generated/Users/UserProfile
+ - Deleted Object: Okta\Sessions/AuthenticationMethod
```
Comment (justinabrokwah-okta) [inline CHANGELOG.md:14]: You used 2 different types of slashes here

File: openapi/generator/cleanFiles.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            }
+        }
+    })
+} catch (e) {
+
+}
+
+fs.writeFile("./generator/createdFiles.json", JSON.stringify(""), function(error) {
+  console.log(error)
+});
```
Comment (justinabrokwah-okta) [inline openapi/generator/cleanFiles.js:31]: Newline at EOF

File: openapi/generator/createdFiles.json [hunk 3]
```diff
@@ -0,0 +1 @@
...iderLinkRequest.php","Policies/UserLifecycleAttributePolicyRuleCondition.php","Users/UserNextLogin.php","Policies/UserPolicyRuleCondition.php","Users/UserProfile.php","Users/UserStatus.php","Policies/UserStatusPolicyRuleCondition.php","UserTypes/UserType.php","UserFactors/VerifyFactorRequest.php","UserFactors/VerifyUserFactorResponse.php","UserFactors/WebAuthnUserFactor.php","UserFactors/WebAuthnUserFactorProfile.php","UserFactors/WebUserFactor.php","UserFactors/WebUserFactorProfile.php","Applications/WsFederationApplication.php","Applications/WsFederationApplicationSettings.php","Applications/WsFederationApplicationSettingsApplication.php","UserFactors/Collection.php","Policies/Collection.php","Users/Collection.php","Applications/Collection.php","AuthorizationServers/Collection.php","Roles/Collection.php","Sessions/Collection.php","Commons/Collection.php","EventHooks/Collection.php","Features/Collection.php","Groups/Collection.php","GroupRules/Collection.php","IdentityProviders/Collection.php","InlineHooks/Collection.php","Ions/Collection.php","LinkedObjects/Collection.php","Logs/Collection.php","Templates/Collection.php","TrustedOrigins/Collection.php","UserTypes/Collection.php"]
```
Comment (justinabrokwah-okta) [inline openapi/generator/createdFiles.json:1]: Newline at EOF

File: src/Applications/ApplicationVisibilityHide.php [hunk 4]
```diff
@@ -1,23 +1,78 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  ******************************************************************************/
 
 namespace Okta\Applications;
 
-class ApplicationVisibilityHide extends \Okta\Generated\Applications\ApplicationVisibilityHide
+class ApplicationVisibilityHide extends \Okta\Resource\AbstractResource
 {
+    const I_OS = 'iOS';
```
Comment (laura-rodriguez) [inline src/Applications/ApplicationVisibilityHide.php:22]: Is `I_OS` the expected name here?

File: src/Applications/CsrMetadata.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
+<?php
+/*******************************************************************************
+ * Copyright 2017 Okta, Inc.
```
Comment (laura-rodriguez) [inline src/Applications/CsrMetadata.php:3]: Should new entities use 2020 copyright?

File: src/Applications/JsonWebKey.php [hunk 6]
```diff
@@ -1,23 +1,216 @@
...
+    const KTY = 'kty';
+    const USE = 'use';
+    const X_5_C = 'x5c';
+    const X_5_T = 'x5t';
+    const X_5_U = 'x5u';
+    const LINKS = '_links';
+    const STATUS = 'status';
+    const CREATED = 'created';
+    const KEY_OPS = 'key_ops';
+    const X_5_T_S_256 = 'x5t#S256';
```
Comment (laura-rodriguez) [inline src/Applications/JsonWebKey.php:35]: Are these the expected names for X5* consts?

File: composer.json [hunk 7]
```diff
@@ -42,14 +42,14 @@
...
-        "rebuild-resources": [
+        "build": [
             "@delete-resources",
             "@generate-resources",
             "@test",
             "@fix-code"
         ],
         "delete-resources": [
-            "rm -rf ./src/Generated"
+            "cd ./openapi && npm run cleanFiles"
```
Comment (laura-rodriguez) [inline composer.json:52]: Should `cleanFiles` be declared as npm script to have this working?

File: src/Applications/CsrMetadata.php (hunk 5 above)
Comment (justinabrokwah-okta) [reply src/Applications/CsrMetadata.php:3]: +1 here, was also wondering

File: openapi/generator/createdFiles.json (hunk 3 above)
Comment (bretterer) [reply openapi/generator/createdFiles.json:1]: This one is auto-generated.

File: src/Applications/CsrMetadata.php (hunk 5 above)
Comment (bretterer) [reply src/Applications/CsrMetadata.php:3]: It should actually be `2017-Present`

File: composer.json (hunk 7 above)
Comment (bretterer) [reply composer.json:52]: No, this calls out from composer.json over to package.json inside of `openapi`

File: src/Applications/JsonWebKey.php (hunk 6 above)
Comment (bretterer) [reply src/Applications/JsonWebKey.php]: Adjusted

File: src/Applications/CsrMetadata.php (hunk 5 above)
Comment (bretterer) [reply src/Applications/CsrMetadata.php]: Adjusted

File: src/Applications/ApplicationVisibilityHide.php (hunk 4 above)
Comment (bretterer) [reply src/Applications/ApplicationVisibilityHide.php]: Adjusted

File: openapi/generator/cleanFiles.js (hunk 2 above)
Comment (bretterer) [reply openapi/generator/cleanFiles.js]: Adjusted

File: CHANGELOG.md (hunk 1 above)
Comment (bretterer) [reply CHANGELOG.md]: Adjusted

**Pull request merged**